src/components/data/TableRow.js

```diff
@@ -0,0 +1,88 @@
+import classnames from 'classnames';
+import { useContext, useEffect } from 'preact/hooks';
+
+import { downcastRef } from '../../util/typing';
+import { useElementRef } from '../../hooks/use-element-ref';
+
+import ScrollContext from './ScrollContext';
+import TableSectionContext from './TableSectionContext';
+import TableContext from './TableContext';
+
+/**
+ * @typedef {import('../../types').PresentationalProps} CommonProps
+ * @typedef {Omit<import('preact').JSX.HTMLAttributes<HTMLElement>, 'size'>} HTMLAttributes
+ *
+ * @typedef TableRowProps
+ * @prop {boolean} [selected=false]
+ */
+
+/**
+ * Render a table row
+ *
+ * @param {CommonProps & TableRowProps & HTMLAttributes} props
+ */
+const TableRowNext = function TableRow({
+  children,
+  classes,
+  elementRef,
+
+  selected,
+
+  ...htmlAttributes
+}) {
+  const rowRef = useElementRef(elementRef);
+
+  const sectionContext = useContext(TableSectionContext);
+  const scrollContext = useContext(ScrollContext);
+  const tableContext = useContext(TableContext);
+
+  const isHeadRow = sectionContext?.section === 'head';
+
+  // Ensure that a selected row is visible when there is a scrolling context
+  useEffect(() => {
+    if (!selected || !scrollContext || !tableContext) {
+      return;
+    }
+    const scrollEl = scrollContext.scrollRef.current;
```

Comment:
In the legacy `Table` implementation, the `Table` component itself renders its own `ScrollBox` and then has to serve as a mediator between that `ScrollBox` and table rows in what feels like a slightly uncomfortable way.

Here, the row can make some decisions about itself based on whether it's in a scroll context, and is able to own its own logic/destiny.